databuilder/query_engines/base_sql.py

```diff
@@ -442,7 +515,7 @@ def get_element_from_value_from_function(
         ]
         return method(*argument_expressions)
 
-    def date_difference_in_years(self, start_date, end_date):
+    def date_difference_in_years(self, start_date: Any, end_date: Any) -> ClauseElement:
```

Comment:
With all the coercion going on here and below, it's a wonder that the type checker makes any sense of it at all. This and the single annotation below seemed like the simplest things we could do.